test/test_mathics_precedence.py

```diff
@@ -0,0 +1,606 @@
+    "Span": 305.0,
+    "SquareIntersection": 305.0,
+    "MinusPlus": 310.0,
+    "PlusMinus": 310.0,
+    "Plus": 310.0,
+    "Subtract": 310.0,
+    "CircleMinus": 330.0,
+    "CirclePlus": 330.0,
+    "Cup": 340.0,
+    "Cap": 350.0,
+    "Coproduct": 360.0,
+    "VerticalTilde": 370.0,
+    "Star": 390.0,
+    "Times": 400.0,
+    "CenterDot": 410.0,
+    "CircleTimes": 420.0,
+    "Vee": 430.0,
+    "Wedge": 440.0,
+    "Diamond": 450.0,
+    "Backslash": 460.0,
+    "Divide": 470.0,
+    "Minus": 480.0,
+    "Dot": 490.0,
+    "CircleDot": 520.0,
+    "SmallCircle": 530.0,
+    "Square": 540.0,
+    "CapitalDifferentialD": 550.0,
+    "Del": 550.0,
+    "DifferentialD": 550.0,
+    "DoubleDownArrow": 580.0,
+    "DoubleLongLeftArrow": 580.0,
+    "DoubleLongLeftRightArrow": 580.0,
+    "DoubleLongRightArrow": 580.0,
+    "DoubleUpArrow": 580.0,
+    "DoubleUpDownArrow": 580.0,
+    "DownArrow": 580.0,
+    "DownArrowBar": 580.0,
+    "DownArrowUpArrow": 580.0,
+    "DownTeeArrow": 580.0,
+    "LeftDownTeeVector": 580.0,
+    "LeftDownVector": 580.0,
+    "LeftDownVectorBar": 580.0,
+    "LeftUpDownVector": 580.0,
+    "LeftUpTeeVector": 580.0,
+    "LeftUpVector": 580.0,
+    "LeftUpVectorBar": 580.0,
+    "LongLeftArrow": 580.0,
+    "LongLeftRightArrow": 580.0,
+    "LongRightArrow": 580.0,
+    "ReverseUpEquilibrium": 580.0,
+    "RightDownTeeVector": 580.0,
+    "RightDownVector": 580.0,
+    "RightDownVectorBar": 580.0,
+    "RightUpDownVector": 580.0,
+    "RightUpTeeVector": 580.0,
+    "RightUpVector": 580.0,
+    "RightUpVectorBar": 580.0,
+    "ShortDownArrow": 580.0,
+    "ShortUpArrow": 580.0,
+    "UpArrow": 580.0,
+    "UpArrowBar": 580.0,
+    "UpArrowDownArrow": 580.0,
+    "UpDownArrow": 580.0,
+    "UpEquilibrium": 580.0,
+    "UpTeeArrow": 580.0,
+    "Power": 590.0,
+    "StringJoin": 600.0,
+    "Factorial": 610.0,
+    "Factorial2": 610.0,
+    "Apply": 620.0,
+    "Map": 620.0,
+    "Prefix": 640.0,
+    "Decrement": 660.0,
+    "Increment": 660.0,
+    "PreDecrement": 660.0,
+    "PreIncrement": 660.0,
+    "Unset": 670.0,
+    "Information": 670.0,
+    "GreaterSlantEqual": 670.0,
+    "LessSlantEqual": 670.0,
+    "Derivative": 670.0,
+    "MapApply": 670.0,
+    "PatternTest": 680.0,
+    "Get": 720.0,
+    "MessageName": 750.0,
+}
+
+SORTED_SYMBOLS_BY_PRECEDENCE = [
+    "CompoundExpression",
+    "Put",
+    "PutAppend",
+    "Set",
+    "SetDelayed",
+    "UpSet",
+    "UpSetDelayed",
+    "Because",
+    "Therefore",
+    "Postfix",
+    "Colon",
+    "Function",
+    "AddTo",
+    "DivideBy",
+    "SubtractFrom",
+    "TimesBy",
+    "ReplaceAll",
+    "ReplaceRepeated",
+    "RuleDelayed",
+    "Rule",
+    "Condition",
+    "StringExpression",
+    "Optional",
+    "Alternatives",
+    "Repeated",
+    "RepeatedNull",
+    "SuchThat",
+    "DoubleLeftTee",
+    "DoubleRightTee",
+    "DownTee",
+    "LeftTee",
+    "Perpendicular",  # 190
+    "RightTee",  # 190
+    # In WMA, `RoundImplies` has a
```

Comment:
We probably need to adjust the  precedence values in the tables for `UpTee` and `RoundImplies`

Reply:
See https://github.com/Mathics3/mathics-scanner/pull/99#discussion_r1863620711